tests/test_adjoint_jacobian.py

```diff
@@ -933,17 +1018,19 @@ def f(params1, params2):
             qml.RY(tf.cos(params2), wires=[0])
             return qml.expval(qml.PauliZ(0))
 
-        if dev.R_DTYPE == np.float32:
+        if dev._new_API:
+            # tf expects float32 with new device API
             tf_r_dtype = tf.float32
+            h = 2e-3
+            tol = 1e-3
```

Comment:
I can't get this test to pass locally without setting this step size and tolerance for the new API. Would appreciate any ideas about why this is the case.

Reply:
Could you please provide more context here? You could push your change to the PR branch so we can see the code and the error message.

Reply:
[Here's a link to the failing CI run](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/8268508294/job/22621563643?pr=607)

Reply:
Discussed offline, we will keep using the "hacky fix" to get the test to pass for now, and will address this in a follow up PR